### ISSUE TITLE
Add GlobalID::Identification for resources passed to jobs

### ISSUE THREAD

**Issue by elrayle** [samvera/hyrax#4099]
### Descriptive summary

If the first parameter to Active Job #perform supports GlobalID::Identification, then Active Job will use call #find(id) on the first parameter to get the full object.  There is a pattern in the jobs that adds a proxy wrapper around a Valkyrie resource to allow it to respond to #find(id).  Currently, the wrapping process needs to be done by the caller.

In this issue, explore the current pattern and determine if the wrapper process can be moved to Hyrax::ApplicationJob.  See 'Current pattern for passing Valkyrie resource to jobs' below for a description of a job that receives a Valkyrie resource.

### Rationale

It is desirable to make this more transparent and easier to work with across all jobs.  

### Current pattern for passing Valkyrie resources to jobs

Job: VisibilityCopyJob #perform(work)

where, work is expected to be an object with GlobalID::Identification mixed in ([ActiveJob Basics - GlobalID](https://guides.rubyonrails.org/active_job_basics.html#globalid)) and can be...
* an ActiveFedora::Base object (which responds to #find(id))
* a Valkyrie::Resource (which does NOT respond to #find(id))

To support enqueueing of the Valkyrie::Resource, it needs to respond to #find(id).  The current implementation of VisibilityCopyJob is tested in [spec/jobs/visibility_copy_job_spec.rb](https://github.com/samvera/hyrax/blob/master/spec/jobs/visibility_copy_job_spec.rb#L5) and uses class [ActiveJobProxy](https://github.com/samvera/hyrax/blob/master/app/models/hyrax/active_job_proxy.rb) to wrap the resource before calling #perform_later.  The proxy provides methods #id and #find(id) which allow Active Job to serialize and deserialize the resource automatically.

To use ActiveJobProxy...

```
proxy_for_resource = Hyrax::ActiveJobProxy.new(resource: resource)
VisibilityCopyJob.perform_later(proxy_for_resource)
```

### Desired pattern

It is desirable to be able to pass the resource to a job without having to wrap it with a proxy.  The exact method for doing this is not yet determined.

It may be as simple as writing a [custom serializer](https://guides.rubyonrails.org/active_job_basics.html#serializers) named something like ValkyrieResourceSerializer and registering it. 

To confirm solution works, the [#perform_later test]((https://github.com/samvera/hyrax/blob/master/spec/jobs/visibility_copy_job_spec.rb#L9-L11) in spec/jobs/visibility_copy_job_spec.rb should succeed using the resource directly without the proxy wrapper.

### Steps to reproduce the behavior

1. Run spec/jobs/visibility_job_spec.rb:9

### Related work

#3877 Support application and propogation of embargos as native Valkyrie

**Comment by no-reply** [comment]
The natural place for this is probably `Hyrax::ApplicationJob.serialize` https://api.rubyonrails.org/classes/ActiveJob/Core.html#method-i-serialize

**Comment by elrayle** [comment]
Ha... turns out the approach I was looking at taking was introduced in Rails 6.  Looking at your suggestion now.

**Comment by elrayle** [comment]
Fixed by PR #4124
